soccer/gameplay/plays/restarts/their_restart.py

```diff
@@ -46,41 +46,16 @@ def execute_running(self):
         their_kicker = min(main.their_robots(),
                            key=lambda opp: opp.pos.dist_to(ball_pos))
 
-        # we build an array of (OpponentRobot, float distToClosestOfOurBots) tuples
-        # we'll use these in a sec to assign our marking robots
-        open_opps_and_dists = []
-        for opp in main.their_robots():
-            # ignore their kicker
-            if opp == their_kicker: continue
-
-            ball_dist = opp.pos.dist_to(ball_pos)
-
-            # see if the opponent is close enough to the ball for us to care
-            # if it is, we record the closest distance from one of our robots to it
-            if ball_dist < 3.0:
-                # which of our robots is closest to this opponent
-                closest_self_dist = min([bot.pos.dist_to(opp.pos)
-                                         for bot in main.our_robots()])
-                open_opps_and_dists.append((opp, closest_self_dist))
+        # we build an array of OpponentRobots sorted rudimentarily by their threat
+        # Right now, this is (inverse of) distance to the ball * 2 + y position.
+        # Needs tuning/improvement. Right now this is excessively defensive
+        sorted_opponents = sorted(
+            filter(lambda x: x != their_kicker, main.their_robots()),
+            key=lambda x: x.pos.dist_to(ball_pos) * 2 + x.pos.y)
```

Comment:
I would almost rename `x` to something else, maybe `r`? Its a little awkward having `x.pos.y`